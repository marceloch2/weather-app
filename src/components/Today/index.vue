<template>
  <div class="today" v-if="weatherData.weather && !hasError && location.city">
    <img
      class="today__img"
      :src=" `http://openweathermap.org/img/w/${weatherData.weather[0].icon}.png`"
    >

    <p class="today__condition">{{ weatherData.weather[0].description }}</p>

    <p class="today__temp">
      <span class="metric" v-html="metric"></span>
      <strong>{{ weatherData.main.temp }}</strong>
    </p>

    <p class="today__wind">
      Wind speed:
      <strong>{{ weatherData.wind.speed }}</strong>
      <span class="metric" v-html="wind_speed"></span>
    </p>

    <p class="today__humidity">
      Humidity:
      <strong>{{ weatherData.main.humidity }}</strong>
      <span class="metric" v-html="humidity"></span>
    </p>

    <p class="today__city">City: {{ location.city }}</p>
  </div>
</template>

<script>
import Today from "./today";
export default Today;
</script>

<style lang="sass" scoped>
	@import 'today.scss'
</style>
