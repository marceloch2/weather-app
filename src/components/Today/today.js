import { mapState } from 'vuex'

export default {
  name: 'today',
  data() {
    return {
      title: 'weather today',
      city: '',
      country: ''
    }
  },

  mounted() {},

  computed: mapState({
    location: state => state.location,
    weatherData: state => state.weatherData_weather,
    metric: state => state.metric,
    wind_speed: state => state.wind_speed,
    humidity: state => state.humidity,
    hasError: state => state.globalError.hasError
  }),

  watch: {
    weatherData: function() {}
  },

  methods: {}
}
