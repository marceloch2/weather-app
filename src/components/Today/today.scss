.today {
  p {
    font-size: 2.2em;
  }

  fieldset {
    border: solid 1px darkslategray;
    padding: 30px;
  }

  input {
    width: 80%;
    padding: 10px;
    margin-bottom: 15px;
    border: solid 1px darkslategray;
  }

  button {
    padding: 10px;
    width: 50%;
    border: none;
    background-color: darkslategray;
    color: #fff;
  }

  .today__condition {
    text-transform: uppercase;
  }

  .today__city {
    text-transform: capitalize;
  }

  .today__country {
    span {
      text-transform: uppercase;
    }
  }
}
