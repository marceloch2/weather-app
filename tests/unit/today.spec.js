import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Today from '@/components/Today'
import weatherData from './mocks/today'

import createStateConfig from './createStateConfig'

const localVue = createLocalVue()
localVue.use(Vuex)

const stateConfig = createStateConfig()

const store = new Vuex.Store({
  state: stateConfig,
  mutations: {}
})

const $router = {
  push: jest.fn()
}

const wrapper = shallowMount(Today, {
  store,
  localVue,
  mocks: {
    $router
  }
})

describe('Today.vue', () => {
  beforeEach(() => {
    wrapper.vm.$store.state.isLoading = false
    wrapper.vm.$store.state.weatherData_weather = weatherData
    wrapper.vm.$store.state.location.city = 'Haarlem'
  })

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Mount with initial state', () => {
    expect(wrapper.vm.title).toMatch('Today Weather')
    expect(wrapper.vm.$store.state.location.city).toBe('Haarlem')
    expect(wrapper.vm.$store.state.location.country).toBe('')
  })

  it('Start with correct UI', () => {
    // Exist and visible
    expect(wrapper.find('.today').isVisible()).toBe(true)
  })

  it('Can render condition', () => {
    expect(wrapper.find('.today__condition').text()).toMatch('mist')
  })

  it('Can render temperature', () => {
    // Temperature
    expect(wrapper.find('.today__temp .metric').text()).toMatch('â„ƒ')
    expect(wrapper.find('.today__temp strong').text()).toMatch('12.8')
  })

  it('Can render wind speed', () => {
    // Wind
    expect(wrapper.find('.today__wind .metric').text()).toMatch('km')
    expect(wrapper.find('.today__wind strong').text()).toMatch('3.6')
  })

  it('Can render humidity', () => {
    // Humidity
    expect(wrapper.find('.today__humidity .metric').text()).toMatch('%')
    expect(wrapper.find('.today__humidity strong').text()).toMatch('66')

    expect(wrapper.find('.today__img').isVisible()).toBe(true)
  })

  it('Can render weather icon', () => {
    expect(wrapper.find('.today__img').isVisible()).toBe(true)
  })
})
