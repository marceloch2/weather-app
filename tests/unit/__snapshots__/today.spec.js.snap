// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Today.vue Snapshot 1`] = `
<div class="today"><img src="http://openweathermap.org/img/w/50n.png" class="today__img">
  <p class="today__condition">mist</p>
  <p class="today__temp"><span class="metric">â„ƒ</span> <strong>12.8</strong></p>
  <p class="today__wind">
    Wind speed:
    <strong>3.6</strong> <span class="metric">km</span></p>
  <p class="today__humidity">
    Humidity:
    <strong>66</strong> <span class="metric">%</span></p>
  <p class="today__city">City: Haarlem</p>
</div>
`;
